### PR TITLE
replace the comparison op for decimal array op using the arrow-rs kernel

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

part of https://github.com/apache/arrow-datafusion/issues/4289

We will replace the arithmetic op later in the follow-up pr.

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Review by alamb — APPROVED** [review]
😍 

Looks great to me -- thanks @liukun4515

File: datafusion/physical-expr/src/expressions/binary/kernels_arrow.rs [hunk 1]
```diff
@@ -516,42 +451,7 @@ mod tests {
             25,
             3,
         );
-        // eq: left == right
```
Comment (liukun4515) [inline datafusion/physical-expr/src/expressions/binary/kernels_arrow.rs:519]: move these test to the test of `binary.rs` file. cc @alamb

File: datafusion/physical-expr/src/expressions/binary.rs [hunk 2]
```diff
@@ -2384,25 +2382,90 @@ mod tests {
         )
         .unwrap();
 
-        // compare decimal array with other array type
+        let value: i128 = 123;
```
Comment (liukun4515) [inline datafusion/physical-expr/src/expressions/binary.rs:2385]: migrate the test in `kernels_arrow.rs` to here

**Comment by alamb** [comment]
CI failure due to https://github.com/apache/arrow-datafusion/issues/4294 (not this PR)

**Pull request merged**